go/vt/vtgate/viperconfig.go

```diff
@@ -1,16 +1,42 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package vtgate
 
-import "vitess.io/vitess/go/viperutil"
+import (
+	vtgatepb "vitess.io/vitess/go/vt/proto/vtgate"
+)
+
+// DynamicViperConfig is a dynamic config that uses viper.
+type DynamicViperConfig struct {
+}
+
+// NewDynamicViperConfig creates a new dynamic viper config
+func NewDynamicViperConfig() *DynamicViperConfig {
+	return &DynamicViperConfig{}
+}
 
-type dynamicViperConfig struct {
-	onlineDDL viperutil.Value[bool]
-	directDDL viperutil.Value[bool]
```

Comment:
@GuptaManan100 I think it's still good to have a local reference here? That makes it easier to test also in the future? We could still provide a function like we have today with `NewDynamicViperConfig` to bind the variables I guess, but it feels a bit more clean to explicitly have it?

Reply:
I don't have strong preference either way, I can introduce the local variables back.